### PR TITLE
Fix shellchecks follow the `${var:?}` pattern

### PR DESCRIPTION
**Pull request by xychu**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug

/kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:
Part of #72956
**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

This is part of fixes under `test/cmd/*.sh` and follows tha pattern in https://github.com/kubernetes/kubernetes/pull/74338

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: test/cmd/storage.sh [hunk 1]
```diff
@@ -27,19 +27,19 @@ run_persistent_volumes_tests() {
...
-  kubectl delete pv pv0001 "${kube_flags[@]}"
-  kubectl create -f test/fixtures/doc-yaml/user-guide/persistent-volumes/volumes/local-02.yaml "${kube_flags[@]}"
-  kube::test::get_object_assert pv "{{range.items}}{{$id_field}}:{{end}}" 'pv0002:'
-  kubectl delete pv pv0002 "${kube_flags[@]}"
-  kubectl create -f test/fixtures/doc-yaml/user-guide/persistent-volumes/volumes/gce.yaml "${kube_flags[@]}"
-  kube::test::get_object_assert pv "{{range.items}}{{$id_field}}:{{end}}" 'pv0003:'
-  kubectl delete pv pv0003 "${kube_flags[@]}"
+  kubectl create -f test/fixtures/doc-yaml/user-guide/persistent-volumes/volumes/local-01.yaml "${kube_flags[@]:?}"
+  kube::test::get_object_assert pv "{{range.items}}{{${id_field:?}}}:{{end}}" 'pv0001:'
+  kubectl delete pv pv0001 "${kube_flags[@]:?}"
```
Comment (ipuustin) [inline test/cmd/storage.sh]: Is the idea that you error out if `kube_flags` is not set or is an empty array? Is that the expected behavior?

Comment (xychu) [reply test/cmd/storage.sh]: Error when not set is what we want here.
So it will pass the shellchecks.

**Comment by xychu** [comment]
@BenTheElder PTAL when it's convenient.

File: test/cmd/storage.sh (hunk 1 above)
Comment (BenTheElder) [reply test/cmd/storage.sh]: empty is a good point, we should confirm that we actually don't expect this to ever be an empty array.

Comment (BenTheElder) [reply test/cmd/storage.sh]: this _looks_ to be true from walking the invoking scripts.

Comment (xychu) [reply test/cmd/storage.sh]: @BenTheElder thanks for your review, do we need to check empty array here, from my tests it seems to be okay.

**Review by BenTheElder — APPROVED** [review]
/lgtm
/approve
will need to go in after code freeze

**Pull request merged**